nautobot/docs/plugins/development.md

```diff
@@ -261,9 +263,92 @@ class SiteAnimalCount(PluginTemplateExtension):
+class DeviceExtraTabs(PluginTemplateExtension):
+    """Template extension to add extra tabs to the object detail tabs."""
+
+    model = 'dcim.device'
+
+    def detail_tabs(self):
+        """
+        You may define extra tabs to render on a model's detail page by utilizing this method.
+        Each tab is defined as a dict in a list of dicts.
+
+        For each of the tabs defined:
+        - The <title> key's value will become the tab link's title.
+        - The <url> key's value is used to render the HTML link for the tab
+
+        These tabs will be visible (in this instance) on the Device model's detail page as
+        set by the DeviceContent.model attribute "dcim.device"
+
+        This example demonstrates defining two tabs. The tabs will be ordered by their position in list.
+        """
+        return [
+            {
+                "title": "Plugin Tab 1",
+                "url": reverse("plugins:example_plugin:device_detail_tab_1", args=[self.context["object"].pk]),
+            },
+            {
+                "title": "Plugin Tab 2",
+                "url": reverse("plugins:example_plugin:device_detail_tab_2", args=[self.context["object"].pk]),
+            },
+        ]
+
+template_extensions = [DeviceExtraTabs, SiteAnimalCount]
+```
+
+#### Adding Extra Tabs
+
+In order for any extra tabs to work properly, the `"url"` key must reference a view which inherits from the `nautobot.core.views.generic.ObjectView` class and the template must extend the object's detail template such as:
+
+```html
+<!-- example_plugin/tab_device_detail_1.html -->
+{% extends 'dcim/device.html' %}
+
+{% block content %}
+    <h2>Device Plugin Tab 1</h2>
+    <p>I am some content for the example plugin's device ({{ object.pk }}) detail tab 1.</p>
+{% endblock %}
 ```
 
+Here's a basic example of a tab's view
+
+```python
+# views.py
+from nautobot.core.views import generic
+from nautobot.dcim.models import Device
+
+class DeviceDetailPluginTabOne(generic.ObjectView):
+    """
+    This view's template extends the device detail template,
+    making it suitable to show as a tab on the device detail page.
+
+    Views that are intended to be for an object detail tab's content rendering must
+    always inherit from nautobot.core.views.generic.ObjectView.
+    """
+
+    queryset = Device.objects.all()
+    template_name = "example_plugin/tab_device_detail_1.html"
+```
+
+You must also add the view to the `url_patterns` like so (make sure to read the note after this code snippet):
+
+```python
+# urls.py
+from django.urls import path
+
+from example_plugin import views
+
+urlpatterns = [
+    # ... previously defined urls
+    path("devices/<uuid:pk>/example-plugin-tab-1/", views.DeviceDetailPluginTabOne.as_view(), name="device_detail_tab_1"),
+]
+```
+
+!!! note 
+    When naming tab view URLs, you may use any naming convention you like, but you must include `<uuid:pk>` in the path string. We recommend using this naming convention:
```

Comment:
I don't think this is correct - you *should* be able to have any parameters in the path that you like so long as you populate them correctly in the `reverse` call in your `detail_tabs()` function. For example, the following *should* work:

```python
    def detail_tabs(self):
        return [
            {
                "title": "Plugin Tab 1",
                "url": reverse("plugins:example_plugin:device_detail_tab_1", kwargs={"slug": self.context["object"].slug}),
            },
        ]

...

urlpatterns = [
    path("sites/<slug:slug>/example-plugin-tab-1/", ...)
]
```

Reply:
See my comment below for a suggested alternate note text.

Reply:
Should I change all the examples to reference the slug instead then?